src/pages/LogView/LoadingPage/useResolveLogURL.ts

```diff
@@ -142,19 +151,32 @@ export const useResolveLogURL = ({
       break;
     }
     case LogTypes.EVERGREEN_TASK_LOGS: {
-      if (!taskID || !execution || !origin) {
+      if (!taskID || !origin || !execution || isLoadingTask) {
         break;
       }
-      downloadURL = getEvergreenTaskLogURL(taskID, execution, origin as any, {
-        priority: true,
-        text: true,
-      });
-      rawLogURL = getEvergreenTaskLogURL(taskID, execution, origin as any, {
-        text: true,
-      });
-      htmlLogURL = getEvergreenTaskLogURL(taskID, execution, origin as any, {
-        text: false,
-      });
+      downloadURL = task?.logs
+        ? getEvergreenTaskLogURL(task.logs, origin, {
+            priority: true,
+            text: true,
+          })
+        : constructEvergreenTaskLogURL(taskID, execution, origin as any, {
+            priority: true,
+            text: true,
+          });
+      rawLogURL = task?.logs
+        ? getEvergreenTaskLogURL(task.logs, origin, {
+            text: true,
+          })
+        : constructEvergreenTaskLogURL(taskID, execution, origin as any, {
+            text: true,
+          });
+      htmlLogURL = task?.logs
+        ? getEvergreenTaskLogURL(task.logs, origin, {
+            text: false,
+          })
+        : constructEvergreenTaskLogURL(taskID, execution, origin as any, {
```

Comment:
wondering if it makes sense to copy the types for `getEvergreenTaskLogURL` to remove the `origin as any` stuff!